packages/flutter/lib/src/rendering/object.dart

```diff
@@ -118,25 +118,33 @@ class PaintingContext extends ClipContext {
       return true;
     }());
     OffsetLayer? childLayer = child._layerHandle.layer as OffsetLayer?;
+    final Offset? oldOffset = childLayer?.offset;
     if (childLayer == null) {
       assert(debugAlsoPaintedParent);
       assert(child._layerHandle.layer == null);
+
       // Not using the `layer` setter because the setter asserts that we not
       // replace the layer for repaint boundaries. That assertion does not
       // apply here because this is exactly the place designed to create a
       // layer for repaint boundaries.
-      final OffsetLayer layer = OffsetLayer();
+      final OffsetLayer layer = child.updateCompositedLayer(oldLayer: null);
       child._layerHandle.layer = childLayer = layer;
     } else {
       assert(debugAlsoPaintedParent || childLayer.attached);
       childLayer.removeAllChildren();
+      final OffsetLayer updatedLayer = child.updateCompositedLayer(oldLayer: childLayer);
+      assert(identical(updatedLayer, childLayer), '$childLayer -> $updatedLayer');
```

Comment:
Make the assert message more descriptive?

Reply:
done